src/h3/mod.rs

```diff
@@ -1257,10 +1257,18 @@ impl Connection {
                     return Err(Error::FrameUnexpected);
                 }
 
+                // Local settings are always set, so this is safe.
+                let max_size = self.local_settings.max_header_list_size.unwrap();
+
                 let headers = self
                     .qpack_decoder
-                    .decode(&mut header_block[..])
-                    .map_err(|_| Error::QpackDecompressionFailed)?;
+                    .decode(&mut header_block[..], max_size as usize)
```

Comment:
Is it safe to assume `max_size` as `usize`? `self.local_settings.max_header_list_size` is `u64` type.
I don't think practically it never exceed 32-bit, but just for clarity.

Reply:
My thinking was that the header list needs to be allocated in memory anyway, so it can't exceed `usize::MAX`, which in turn means that truncating the `max_size` value to that should be safe (the main annoyance with using `u64` is that we would need to cast `usize` values in the `decode()` function 3 times, rather than do a single `u64` -> `usize` cast here).

Reply:
Sounds good. This `usize` is platform dependent keep annoying me :)

Reply:
So, thinking about this some more, it's probably just safer to use `u64` everywhere even if it's slightly more annoying, so I changed this to use `u64` instead of `usize` now.